File: examples/javascript/chat_example.js
/**
 * OpenAI Compatible API - Multi-Model Chat (Node.js)
 * Use GPT-4, Claude, Gemini with OpenAI SDK via Mountsea AI
 *
 * Documentation: https://docs.mountsea.ai/api-reference/chat/introduction
 * Platform: https://shanhaiapi.com/zh/
 */

import OpenAI from 'openai';

const client = new OpenAI({
  apiKey: process.env.MOUNTSEA_API_KEY || 'your-api-key',
  baseURL: 'https://api.mountsea.ai/v1'
});

async function chatWith(model, message) {
  const response = await client.chat.completions.create({
    model,
    messages: [{ role: 'user', content: message }]
  });
  return response.choices[0].message.content;
}

async function streamChat(model, message) {
  const stream = await client.chat.completions.create({
    model,
    messages: [{ role: 'user', content: message }],
    stream: true
  });
  for await (const chunk of stream) {
    process.stdout.write(chunk.choices[0]?.delta?.content || '');
  }
  console.log();
}

(async () => {
  console.log('🤖 GPT-4:', await chatWith('gpt-4', 'Hello!'));
  console.log('🟣 Claude:', await chatWith('claude-3-opus', 'Hello!'));
  console.log('🔵 Gemini:', await chatWith('gemini-pro', 'Hello!'));
  console.log('\n📡 Streaming:');
  await streamChat('gpt-4', 'Tell me a joke');
  console.log('\n📘 Docs: https://docs.mountsea.ai/api-reference/chat/introduction');
})();

File: examples/python/chat_example.py
"""
OpenAI Compatible API - Chat Example
Use any OpenAI SDK to access GPT-4, Claude, Gemini and more

Documentation: https://docs.mountsea.ai/api-reference/chat/introduction
Platform: https://shanhaiapi.com/zh/
"""

from openai import OpenAI
import os

API_KEY = os.environ.get("MOUNTSEA_API_KEY", "your-api-key")

# Just change the base_url to Mountsea AI
client = OpenAI(
    api_key=API_KEY,
    base_url="https://api.mountsea.ai/v1"
)


def chat_with_gpt4(message: str) -> str:
    """Chat using GPT-4."""
    response = client.chat.completions.create(
        model="gpt-4",
        messages=[
            {"role": "system", "content": "You are a helpful assistant."},
            {"role": "user", "content": message}
        ],
        temperature=0.7,
        max_tokens=1000
    )
    return response.choices[0].message.content


def chat_with_claude(message: str) -> str:
    """Chat using Claude via OpenAI-compatible API."""
    response = client.chat.completions.create(
        model="claude-3-opus",
        messages=[
            {"role": "user", "content": message}
        ]
    )
    return response.choices[0].message.content


def chat_with_gemini(message: str) -> str:
    """Chat using Gemini via OpenAI-compatible API."""
    response = client.chat.completions.create(
        model="gemini-pro",
        messages=[
            {"role": "user", "content": message}
        ]
    )
    return response.choices[0].message.content


def streaming_chat(message: str, model: str = "gpt-4"):
    """Streaming chat example."""
    stream = client.chat.completions.create(
        model=model,
        messages=[{"role": "user", "content": message}],
        stream=True
    )
    
    full_response = ""
    for chunk in stream:
        if chunk.choices[0].delta.content:
            content = chunk.choices[0].delta.content
            print(content, end="", flush=True)
            full_response += content
    
    print()
    return full_response


if __name__ == "__main__":
    # GPT-4 Example
    print("🤖 GPT-4:")
    print(chat_with_gpt4("What is artificial intelligence?"))
    print()
    
    # Claude Example
    print("🟣 Claude:")
    print(chat_with_claude("Write a haiku about coding."))
    print()
    
    # Gemini Example
    print("🔵 Gemini:")
    print(chat_with_gemini("Explain machine learning in one sentence."))
    print()
    
    # Streaming Example
    print("📡 Streaming (GPT-4):")
    streaming_chat("Tell me a short joke about programming.")
    
    print("\n🔗 Documentation: https://docs.mountsea.ai/api-reference/chat/introduction")
    print("🏠 Platform: https://shanhaiapi.com/zh/")

